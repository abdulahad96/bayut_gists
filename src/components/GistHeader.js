import React from 'react'
import styled from 'styled-components';
import Octicon from 'react-octicon'
import PropTypes from 'prop-types';
import LabelIcon from './LabelIcon';

const Container = styled.div`
flex-direction:row;
display:flex;
flex:1;
color:#126ed8;
font-weight:bold;
width:100%`;

const NameContainer = styled.a`
justify-content: flex-start;
align-items: center;
display: flex;
flex:6;
flex-direction: row;
gap: 5px;
text-decoration: none;

`;
const NameText = styled.div`
border-radius: 4px;
display:flex;
font-size:14px;
flex-direction:row;
`;
const Image = styled.img`
border-radius:50px;
height:35px;
width:35px
`;
const InfoContainer = styled.div`
flex:4.5;
display:flex;
justify-content:space-between;
flex-direction:row;
gap:10px;

`;

const GistHeader = ({owner,file,comments_url,git_pull_url,url,forks_url,html_url}) => {
    const {avatar_url,login} = owner;
    const {files_items} =file
    return (
        <Container>
            <NameContainer href={url}>
                <Image src={avatar_url} />
                <NameText>{login}</NameText>
            </NameContainer>
            <InfoContainer>
                <LabelIcon
                name={'code'} Url={html_url} Txt={`${files_items.length} files`}
                />
                <LabelIcon name={'repo-forked'} Url={forks_url} Txt="Fork"/>
                <LabelIcon name={'comment'} Url={comments_url} Txt={"Comment"}/>
            <LabelIcon name={'repo-pull'} Url={git_pull_url} Txt={"Pull"}/>
            </InfoContainer>
        </Container>

    )
}

GistHeader.propTypes = {
    owner: PropTypes.shape({
      avatar_url: PropTypes.string.isRequired,
      login: PropTypes.string.isRequired,
    }).isRequired,
    file: PropTypes.shape({
      files_items: PropTypes.array.isRequired,
    }).isRequired,
    comments_url: PropTypes.string.isRequired,
    git_pull_url: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    forks_url: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired,
  };
export default GistHeader;