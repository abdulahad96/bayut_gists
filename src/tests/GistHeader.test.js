import React from 'react';
import { render } from '@testing-library/react';
import GistHeader from '../components/GistHeader';

describe('GistHeader', () => {
    const   owner= {
        login: "danbahrami",
        avatar_url: "https://avatars.githubusercontent.com/u/6588325?v=4",
        }
        const url = "https://avatars.githubusercontent.com/u/6588325?v=4";
      const html_url="https://avatars.githubusercontent.com/u/6588325?v=4";
     const git_pull_url="https://avatars.githubusercontent.com/u/6588325?v=4"
     const forks_url="https://avatars.githubusercontent.com/u/6588325?v=4"
     const comments_url = "https://avatars.githubusercontent.com/u/6588325?v=4"
      const  file={files:{
            "resume.json": {
                 filename: "resume.json",
                 type: "application/json",
                 language: "JSON",
                 raw_url: "https://gist.githubusercontent.com/danbahrami/f691b3fc6a507980a4b5a2fbca2b1498/raw/6292ab1d5c7e58bc09f5d00eb970a66cfa329769/resume.json",
                 size: 11202
             }
         } , files_items:['resume.json']}
    
  test('renders Gistheader correctly', () => {
    const output = render(
        <GistHeader
        owner={owner}
        url = {url}
        html_url={html_url}
        git_pull_url={git_pull_url}
        forks_url={forks_url}
        file={file}
        comments_url={comments_url}
    />
    );
    expect(output).toMatchSnapshot();
  });

});
