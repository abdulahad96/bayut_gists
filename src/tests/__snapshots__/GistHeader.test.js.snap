// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GistHeader renders Gistheader correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-eqUAAy haWCne"
      >
        <a
          class="sc-fqkvVR gFoZWS"
          href="https://avatars.githubusercontent.com/u/6588325?v=4"
        >
          <img
            class="sc-iGgWBj eELUQN"
            src="https://avatars.githubusercontent.com/u/6588325?v=4"
          />
          <div
            class="sc-dcJsrY gqanGQ"
          >
            danbahrami
          </div>
        </a>
        <div
          class="sc-gsFSXq hNDJdp"
        >
          <a
            class="sc-aXZVg fUzSPB"
            href="https://avatars.githubusercontent.com/u/6588325?v=4"
          >
            <span
              class="octicon octicon-code"
            />
            <div
              class="sc-gEvEer hydGKF"
            >
              1 files
            </div>
          </a>
          <a
            class="sc-aXZVg fUzSPB"
            href="https://avatars.githubusercontent.com/u/6588325?v=4"
          >
            <span
              class="octicon octicon-repo-forked"
            />
            <div
              class="sc-gEvEer hydGKF"
            >
              Fork
            </div>
          </a>
          <a
            class="sc-aXZVg fUzSPB"
            href="https://avatars.githubusercontent.com/u/6588325?v=4"
          >
            <span
              class="octicon octicon-comment"
            />
            <div
              class="sc-gEvEer hydGKF"
            >
              Comment
            </div>
          </a>
          <a
            class="sc-aXZVg fUzSPB"
            href="https://avatars.githubusercontent.com/u/6588325?v=4"
          >
            <span
              class="octicon octicon-repo-pull"
            />
            <div
              class="sc-gEvEer hydGKF"
            >
              Pull
            </div>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-eqUAAy haWCne"
    >
      <a
        class="sc-fqkvVR gFoZWS"
        href="https://avatars.githubusercontent.com/u/6588325?v=4"
      >
        <img
          class="sc-iGgWBj eELUQN"
          src="https://avatars.githubusercontent.com/u/6588325?v=4"
        />
        <div
          class="sc-dcJsrY gqanGQ"
        >
          danbahrami
        </div>
      </a>
      <div
        class="sc-gsFSXq hNDJdp"
      >
        <a
          class="sc-aXZVg fUzSPB"
          href="https://avatars.githubusercontent.com/u/6588325?v=4"
        >
          <span
            class="octicon octicon-code"
          />
          <div
            class="sc-gEvEer hydGKF"
          >
            1 files
          </div>
        </a>
        <a
          class="sc-aXZVg fUzSPB"
          href="https://avatars.githubusercontent.com/u/6588325?v=4"
        >
          <span
            class="octicon octicon-repo-forked"
          />
          <div
            class="sc-gEvEer hydGKF"
          >
            Fork
          </div>
        </a>
        <a
          class="sc-aXZVg fUzSPB"
          href="https://avatars.githubusercontent.com/u/6588325?v=4"
        >
          <span
            class="octicon octicon-comment"
          />
          <div
            class="sc-gEvEer hydGKF"
          >
            Comment
          </div>
        </a>
        <a
          class="sc-aXZVg fUzSPB"
          href="https://avatars.githubusercontent.com/u/6588325?v=4"
        >
          <span
            class="octicon octicon-repo-pull"
          />
          <div
            class="sc-gEvEer hydGKF"
          >
            Pull
          </div>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
